src/app/dim-api/import.ts

```diff
@@ -149,16 +150,20 @@ export function importDataBackup(data: ExportResponse, silent = false): ThunkRes
 }
 
 /** Returns a promise that resolves when the profile is fully loaded. */
-function waitForProfileLoad() {
+function waitForProfileLoad<D extends Dispatch>(dispatch: D) {
   return new Promise((resolve) => {
-    const unsubscribe = observeStore(
-      (state) => state.dimApi.profileLoaded,
-      (_prev, loaded) => {
-        if (loaded) {
-          unsubscribe();
-          resolve(undefined);
-        }
-      },
+    const id = 'profile-loaded-observer';
+    dispatch(
+      observe({
+        id,
+        getObserved: (state) => state.dimApi.profileLoaded,
+        sideEffect: ({ current }) => {
```

Comment:
I think this one we want to have as a "run initially" in the event that the profile is loaded after this is dispatched but before it is received.